test/khash_test.erl

```diff
@@ -0,0 +1,443 @@
+                    khash:put(H, foo, bar2),
+                    {
+                        "put expires iterators",
+                        ?_assertEqual(khash:iter_next(I), Err)
+                    }
+                end,
+                fun({ok, H, I, Err}) ->
+                    khash:del(H, foo),
+                    {
+                        "del expires iterators",
+                        ?_assertEqual(khash:iter_next(I), Err)
+                    }
+                end,
+                fun({ok, H, I, Err}) ->
+                    khash:clear(H),
+                    {
+                        "clear expires iterators",
+                        ?_assertEqual(khash:iter_next(I), Err)
+                    }
+                end
+            ]
+        }
+    }.
+
+no_expiration_iterators_test_() ->
+    {
+        "khash iterators no exipiration functions",
+        {foreach,
+            local,
+            fun() ->
+                Err = {error, expired_iterator},
+                {ok, H} = khash:new(),
+                khash:put(H, foo, bar),
+                {ok, I} = khash:iter(H),
+                {ok, H, I, Err}
+            end,
+            [
+                fun({ok, H, I, Err}) ->
+                    [{foo, bar}] = khash:to_list(H),
+                    {
+                        "to_list doesn't expire iterators",
+                        ?_assertNot(khash:iter_next(I) == Err)
+                    }
+                end,
+                fun({ok, H, I, Err}) ->
+                    {value, bar} = khash:lookup(H,foo),
+                    {
+                        "lookup doesn't expire iterators",
+                        ?_assertNot(khash:iter_next(I) == Err)
+                    }
+                end,
+                fun({ok, H, I, Err}) ->
+                    bar = khash:get(H, foo),
+                    {
+                        "get doesn't expire iterators",
+                        ?_assertNot(khash:iter_next(I) == Err)
+                    }
+                end,
+                fun({ok, H, I, Err}) ->
+                    1 = khash:size(H),
+                    {
+                        "size doesn't expire iterators",
+                        ?_assertNot(khash:iter_next(I) == Err)
+                    }
+                end,
+                fun({ok, H, I, Err}) ->
+                    {ok, I} = khash:iter(H),
+                    {foo, bar} = khash:iter_next(I),
+                    end_of_table = khash:iter_next(I),
+                    {
+                        "iteration doesn't expire iterators",
+                        ?_assertNot(khash:iter_next(I) == Err)
+                    }
+                end
+            ]
+        }
+    }.
+
+
+dict_fetch() ->
+    erlang:garbage_collect(),
+    List = kv_data(?NUM_KVS),
+    Dict = dict:from_list(List),
+    dict_fetch(Dict, ?NUM_CYCLES * 10).
+
+dict_fetch(_, 0) ->
+    ok;
+dict_fetch(Dict, NumCycles) ->
+    1 = dict:fetch(1, Dict),
```

Comment:
Those are actually not assertions, but helpers for basic perf test of khash vs dict, i.e. we are not doing functional testing for dict of khash fetch here. But sure, no harm in converting them to the asserts, I'll update that.